sweep/sweeper.go

```diff
@@ -1,24 +1,59 @@
 package sweep
 
 import (
+	"errors"
+	"fmt"
 	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/txscript"
 	"github.com/btcsuite/btcd/wire"
 	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/chainntnfs"
 	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/ticker"
+	"sync"
+	"sync/atomic"
+	"time"
 )
 
-// UtxoSweeper provides the functionality to generate sweep txes. The plan is
-// to extend UtxoSweeper in the future to also manage the actual sweeping
-// process by itself.
+var (
+	// DefaultSweepCheckInterval specifies the time intervals between checks
+	// for inputs to sweep.
+	DefaultSweepCheckInterval = time.Second
+
+	// DefaultBatchWindowSize specifies the number of sweep check intervals
+	// to pass without new inputs being added before the sweep is published.
+	DefaultBatchWindowSize = 30
```

Comment:
Alternatively we can just use a single value here?

Reply:
If we use a single value, the risk exists that the tick happens right after the block epoch and inputs are still being fed into the sweeper. It will create two sweeps that could have been one. That's why I have the two values.

Your suggestion below of resetting the ticker solves the problem too, only Ticker has no reset function on it. I didn't want to change Ticker, but we could decide otherwise.

Reply:
I also looked at `time.Timer`. I would say it fits the purpose much better. But reading the details on `Stop()` and `Reset()` discouraged me. Looks not so easy to get management of in-flights sends on the channel `C` correct. Do you have experience with this one?

Reply:
For timers, you can just create a new one each time instead of resetting.

Reply:
Ok, good suggestion. Will try it out.